tools/hlstool/hlstool.cpp

```diff
@@ -119,40 +119,67 @@ static cl::opt<HLSFlow>
                                   "Statically scheduled (Calyx path)")),
             cl::cat(mainCategory));
 
-enum DynamicParallelismKind {
-  DynamicParallelismNone,
-  DynamicParallelismLocking,
-  DynamicParallelismPipelining
-};
+enum DynamicParallelismKind { None, Locking, Pipelining };
 
 static cl::opt<DynamicParallelismKind> dynParallelism(
     "dynamic-parallelism", cl::desc("Specify the DHLS task parallelism kind"),
     cl::values(
-        clEnumValN(DynamicParallelismNone, "none",
+        clEnumValN(None, "none",
                    "Add no protection mechanisms that could prevent data races "
                    "when a function has multiple active invocations."),
-        clEnumValN(DynamicParallelismLocking, "locking",
+        clEnumValN(Locking, "locking",
                    "Add function locking protection mechanism which ensures "
                    "that only one function invocation is active."),
-        clEnumValN(DynamicParallelismPipelining, "pipelining",
+        clEnumValN(Pipelining, "pipelining",
                    "Add function pipelining mechanism that enables a "
                    "pipelined execution of multiple function invocations while "
                    "preserving correctness.")),
-    cl::init(DynamicParallelismPipelining), cl::cat(mainCategory));
+    cl::init(Pipelining), cl::cat(mainCategory));
 
-enum OutputFormatKind { OutputIR, OutputVerilog, OutputSplitVerilog };
+enum IRLevel {
```

Comment:
Similar here

Reply:
Ah, unfortunately there is code that relies on interpreting the variants in the enum as integers to suppress unneeded passes. It is possible to convert this stuff into `enum class` but would require more changes that are not directly relevant to this so I can do them in a follow PR if that's okay?

Reply:
Sure, if you don't plan on doing it right away definitely add an issue for it. This would allow users to pass `None` instead of `MyEnumName::None` or previously `MyEnumNameNone`, which could lead to tricky bugs.